src/empty.js

```diff
@@ -26,9 +27,10 @@ module.exports = _curry1(function empty(x) {
     return x.constructor.empty();
   } else {
     switch (Object.prototype.toString.call(x)) {
-      case '[object Array]':  return [];
-      case '[object Object]': return {};
-      case '[object String]': return '';
+      case '[object Arguments]':  return (function() { return arguments; }());
```

Comment:
Here's Underscore's [`_.isArguments` fallback][1]:

> ```javascript
> // Define a fallback version of the method in browsers (ahem, IE < 9), where
> // there isn't any inspectable "Arguments" type.
> if (!_.isArguments(arguments)) {
>   _.isArguments = function(obj) {
>     return _.has(obj, 'callee');
>   };
> }
> ```

If it's IE8 as well, we should probably provide the same fallback.


[1]: https://github.com/jashkenas/underscore/blob/1.8.3/underscore.js#L1224-L1230